code-challenges.test.js

```diff
@@ -14,33 +17,148 @@
+// Process: create a function called peopleList and pass in a parameter of an array. 
+// Create another variable called newList that calls in a new function that will store the array that has been iterated through using .map ? I tried several ways to figure out how to get the array.map to store to a function, the internet search suggested to use the arrow syntax
+// in so that we can return a terate through the function using a .map method
+//create a variable that will store a new array use a method to take each key value pair and return them as a sentence ? using the .split method
+//then add the .map method again to capitilize the word at the 0 index
+//use a method to capitilize a the first word in the sentence in the array
+//use the method .join to turn into array once more
+//return 
+
+
+
+// const peopleList = (array) => {
+//   let newList => { 
+//     array.map 
+// } newList = array.split('').map(array[0].toUpperCase().join())
+// }
+
+// console.log(peopleList(hitchhikersCharacters))
+
+
+
 
 // --------------------2) Create a function that takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.
 
 // a) Create a test with an expect statement using the variables provided.
 
-const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
+//const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
 // Expected output: [ 2, 0, -1, 0 ]
-const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
+// const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
 // Expected output: [ 2, 1, -1 ]
 
+// describe("remainderList", () => {
+//     it("creates a function that takes a mixed data array and returns an array of only the remainders of the number divided by 3", () => {
+//     expect(remainderList(hodgepodge1)).toEqual([ 2, 0, -1, 0 ])
+//     expect(remainderList(hodgepodge2)).toEqual([ 2, 1, -1 ])
+//  })
+// })
+
+//RED - Good failure
+// FAIL  ./code-challenges.test.js
+// remainderList
+//   ✕ creates a function that takes a mixed data array and returns an array of only the remainders of the number divided by 3 (1 ms)
+
+// ● remainderList › creates a function that takes a mixed data array and returns an array of only the remainders of the number divided by 3
+
+//   ReferenceError: remainderList is not defined
+
+
 // b) Create the function that makes the test pass.
+//PseudoCode
+//Input: const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
+// const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
+//Output: [ 2, 0, -1, 0 ], [ 2, 1, -1 ]
+//Process: Create a function aclled remainderList and pass in array as a parameter
+//use the .filter method to filter through the array to find the data type of number, specify data type for the target data type using 'typeof', specify number
+//map through the array to find the data type from above and divide by 3
+//output is undefined
```

Comment:
Good logic here! Output is undefined because you don't have a return in your function